system/jlib/jmetrics.cpp

```diff
@@ -162,43 +163,69 @@ bool MetricsManager::addMetric(const std::shared_ptr<IMetric> &pMetric)
         {
             auto metaData = pMetric->queryMetaData();
             std::unique_lock<std::mutex> lock(metricVectorMutex);
+            bool metaDataOk = true;
             for (auto &metaDataIt: metaData)
             {
-                name.append(".").append(metaDataIt.value);
-            }
-
-            auto it = metrics.find(name);
-            if (it == metrics.end())
-            {
-                metrics.insert({name, pMetric});
-                rc = true;
-            }
-            else
-            {
-                // If there is a match only report an error if the metric has not been destroyed in the meantime
-                auto match = it->second.lock();
-                if (match)
+                try
                 {
+                    if (std::regex_match(metaDataIt.value, metaDataValidator))
+                    {
+                        name.append(".").append(metaDataIt.value);
+                    }
+                    else
+                    {
+                        metaDataOk = false;
 #ifdef _DEBUG
-                    // In debug throw an exception so the developer knows when a duplicate metric name is being added
-                    throw makeStringExceptionV(MSGAUD_operator, "addMetric - Attempted to add duplicate named metric with name '%s'", name.c_str());
+                        // In debug throw an exception to notify the developer of the invalid metadata value
+                        throw makeStringExceptionV(MSGAUD_operator, "addMetric - Attempted to add metric with invalid metadata '%s' for metric '%s'", metaDataIt.value.c_str(), name.c_str());
 #else
-                    // In release notify the operator of the error, but don't prevent the system from loading
-                    OERRLOG("addMetric - Adding a duplicate named metric '%s', old metric replaced", name.c_str());
+                        // In release notify the operator of the error, but don't prevent the system from loading
+                        OERRLOG("addMetric - Metric metadata '%s' is not valid for metric '%s', metric not added", metaDataIt.value.c_str(), name.c_str());
+                        break;
 #endif
+                    }
+                }
+                catch (std::regex_error &regex_error)
```

Comment:
this inner try/catch seems superfluous

Reply:
Agreed. Inner try/catch removed. Outer block will handle it.